### PR TITLE
feat(prqlc): add the `prqlc shell-completion` command to generate shell completions

### PR DESCRIPTION
**Pull request by eitsupi**

### CODE REVIEW

File: prql-compiler/prqlc/tests/test.rs [hunk 1]
```diff
@@ -75,3 +76,1145 @@ fn test_compile() {
     ----- stderr -----
     "###);
 }
+
+#[test]
+fn test_shell_completion() {
+    assert_cmd_snapshot!(Command::new(get_cargo_bin("prqlc")).arg("shell-completion").arg("bash"), @r###"
```
Comment (max-sixty) [inline prql-compiler/prqlc/tests/test.rs]: Possibly given the length of this, we don't put inline? If we remove the `@""`, insta makes a snapshot file.

**Review by max-sixty — APPROVED** [review]
Awesome!! Very cool.

One idea is to put the instructions into `--help` rather / in addition to the docs, so they're closer to their usage. But not sure how viable that is.

And a separate follow-up might be to add this to the brew recipe so it gets added automatically.

**Comment by eitsupi** [comment]
Thanks for your review!

I think we can improve this more in the future as like your idea, but for now I would like to merge this and allow others to work on that.

File: prql-compiler/prqlc/tests/test.rs (hunk 1 above)
Comment (eitsupi) [reply prql-compiler/prqlc/tests/test.rs]: Thanks, I didn't know that!

**Pull request merged**